File: Server/schemas/convalidation/convalidation_out.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from utils.constants import MAX_LENGTH_NAME, MAX_LENGTH_EMAIL

class ConvalidationOut(BaseModel):
    id_convalidation: int
    review_comments: str | None
    id_request: int
    id_convalidation_type: int
    id_convalidation_state: int
    id_curriculum_course: int
    # Datos de la solicitud
    sent_at: datetime
    reviewed_at: datetime | None
    id_student: int
    id_reviewed_by: int | None
    # Datos del estudiante
    rol_student: str
    rut_student: str
    student_name: str = Field(..., max_length=MAX_LENGTH_NAME)
    student_campus: str
    student_email: str = Field(..., max_length=MAX_LENGTH_EMAIL)
    # Datos del revisor
    reviewer_id: int | None
    reviewer_name: str | None = Field(None, max_length=MAX_LENGTH_NAME)
    reviewer_campus: str | None
    reviewer_email: str | None = Field(None, max_length=MAX_LENGTH_EMAIL)
    # Datos de tipos y estados
    convalidation_type: str = Field(..., max_length=MAX_LENGTH_NAME)
    convalidation_state: str = Field(..., max_length=MAX_LENGTH_NAME)
    curriculum_course: str = Field(..., max_length=MAX_LENGTH_NAME)
    curriculum_course_type: str = Field(..., max_length=MAX_LENGTH_NAME) 

File: Client/src/features/convalidations/types/convalidation_out.ts
// convalidation_out.ts
// Interface de convalidación (output) basada en ConvalidationOut del backend


export interface Convalidation {
  id_request: number
  id_student: number
  sent_at?: string // datetime
  reviewed_at?: string // datetime
  student_name: string
  student_rut: string
  student_rol: string
  student_campus: string
  id_reviewed_by?: number
  reviewed_by?: string
  id_convalidation: number
  review_comments?: string
  id_convalidation_type: number
  convalidation_type: string
  id_convalidation_state: number
  convalidation_state: string
  id_curriculum_course: number
  curriculum_course: string
}

export interface ConvalidationSubjectOut extends Convalidation {
  id_subject: number
  subject: string
  department: string
}

export interface ConvalidationWorkshopOut extends Convalidation {
  id_convalidation_workshop: number
  id_convalidation: number
  id_workshop: number
  workshop: string
  semester: string
  year: number
}

export interface ConvalidationExternalActivityOut extends Convalidation {
  id_convalidation_external_activity: number
  id_convalidation: number
  id_external_activity: number
  external_activity: string
  semester: string
  year: number
}
  
export interface ConvalidationOut {
  subject: [ConvalidationSubjectOut]
  workshop: ConvalidationWorkshopOut
  external_activity: ConvalidationExternalActivityOut
}